### PR TITLE
add enrollment models

### PR DESCRIPTION
**Pull request by abeglova**
#### Pre-Flight checklist
- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
closes https://github.com/mitodl/open-discussions/issues/3297

#### What's this PR do?
Adds a table to store enrollments in open

#### How should this be manually tested?
Make sure the migration runs

### CODE REVIEW

**Comment by pdpinch** [comment]
👍 Schema looks OK to me.

**Review by mbertrand — CHANGES_REQUESTED** [review]
Getting an error on startup: "module 'enrollments' has no attribute 'default_app_config" - adding an `__init__.py` file to the app should fix this.

Might be useful to add the Enrollment model to the Django Admin page.

I probably would have placed this model in the existing `course_catalog` app, but a separate app works too and maybe makes more sense if there's lots more functionality coming (tasks, views?) that's centered around these enrollments.

**Comment by abeglova** [comment]
I moved the model to course_catalog and added an admin view

File: course_catalog/admin.py [hunk 1]
```diff
@@ -145,6 +146,15 @@ class PodcastEpisodeAdmin(admin.ModelAdmin):
     search_fields = ("full_description",)
 
 
+class EnrollmentAdmin(admin.ModelAdmin):
+    """Enrollment Admin"""
+
+    model = Enrollment
+    search_fields = ("run_id", "user_id")
+    list_display = ("run_id", "user_id", "enrollment_timestamp")
+    search_fields = ("run_id", "user_id")
```
Comment (mbertrand) [inline course_catalog/admin.py:155]: `search_fields` occurs twice.  Depending on whether or not Peter/Brian think they might need to use this, might be worth making it more user-friendly by doing something like this:

```python
    def run_slug(self, obj):
        return f"{obj.run.slug}"

    run_slug.admin_order_field = 'run__slug'

    model = Enrollment
    search_fields = ("run__run_id", "run__slug", "run__title", "user__email", "user__username")
    list_display = ("run_slug", "user", "enrollment_timestamp")
    autocomplete_fields = ("run", "user")
```

**Review by mbertrand — CHANGES_REQUESTED** [review]

File: course_catalog/admin.py (hunk 1 above)
Comment (abeglova) [reply course_catalog/admin.py]: Thanks! I updated the code to use your suggestion

**Review by mbertrand — APPROVED** [review]
👍

**Pull request merged**